scripts/zones/Nyzul_Isle/instances/nyzul_isle_investigation.lua

```diff
@@ -0,0 +1,182 @@
+-----------------------------------
+
+function afterInstanceRegister(player)
+    local instance = player:getInstance()
+    player:messageSpecial(ID.text.TIME_TO_COMPLETE, instance:getTimeLimit())
+end
+
+function onInstanceCreated(instance)
+end
+
+function onInstanceTimeUpdate(instance, elapsed)
+    updateInstanceTime(instance, elapsed, ID.text)
+end
+
+function onInstanceFailure(instance)
+    local chars = instance:getChars()
+
+    for i,v in ipairs(chars) do
+        v:messageSpecial(ID.text.MISSION_FAILED,10,10)
+        v:startEvent(1)
+    end
+end
+
+function onInstanceProgressUpdate(instance, progress)
+    local chars = instance:getChars()
+    local stage = instance:getStage()
+
+    if stage == (nyzul.objective.FREE_FLOOR or stage == nyzul.objective.ELIMINATE_ENEMY_LEADER or stage == nyzul.objective.ELIMINATE_SPECIFIED_ENEMY) and progress == 15 then
+        instance:getEntity(bit.band(ID.npc.RUNE_TRANSFER, 0xFFF), dsp.objType.NPC):AnimationSub(1)
+        for i,v in ipairs(chars) do
+            v:messageSpecial(ID.text.OBJECTIVE_COMPLETE, instance:getEntity(bit.band(ID.npc.RUNE_TRANSFER_START, 0xFFF), dsp.objType.NPC):getLocalVar("[Nyzul_Current_Floor"))
+        end
+    end
+
+end
+
+function onInstanceComplete(instance)
+
+end
+
+function pickSetPoint(player)
+    local instance = player:getInstance()
+    local chars = instance:getChars()
+    local START = instance:getEntity(bit.band(ID.npc.RUNE_TRANSFER_START, 0xFFF), dsp.objType.NPC)
+    local TRANSFER = instance:getEntity(bit.band(ID.npc.RUNE_TRANSFER, 0xFFF), dsp.objType.NPC)
+    local FLOOR = START:getLocalVar("[Nyzul_Current_Floor")
+    
+    instance:setProgress(10)
+    if math.random(1,30) == 10 then -- 3.33% of a free floor
+        instance:setStage(nyzul.objective.FREE_FLOOR)
+        TRANSFER:timer(3200, function(npc) instance:setProgress(15) end)
+    elseif FLOOR == 20 or FLOOR == 40 or FLOOR == 60 or FLOOR == 80 then
+        instance:setStage(nyzul.objective.ELIMINATE_ENEMY_LEADER)
+        START:setLocalVar("[Nyzul_Isle]FloorLayout", 0)
+    else
+        instance:setStage(math.random(nyzul.objective.ELIMINATE_ENEMY_LEADER, nyzul.objective.ELIMINATE_ALL_ENEMIES))
+    end
+    for k, v in pairs(nyzul.FloorLayout) do
+        if k == START:getLocalVar("[Nyzul_Isle]FloorLayout") then
+            TRANSFER:setPos(v[1], v[2], v[3])
+            TRANSFER:AnimationSub(0)
+            pickMobs(player)
+            local posX = v[1] local posY = v[2] local posZ = v[3]
+            for i,v in pairs(chars) do
+                v:timer(3200, function(player)
+                v:setPos(posX + math.random(-2, 2), posY, posZ + math.random(-2, 2))
+                v:messageSpecial(ID.text.WELCOME_TO_FLOOR, FLOOR) end)
+                if instance:getStage() ~= nyzul.objective.FREE_FLOOR then
+                    v:timer(3200, function(player)
+                    v:messageSpecial(ID.text.ELIMINATE_ENEMY_LEADER + instance:getStage()) end)
```

Comment:
Lamp objectives are on the lamps themselves, not on the rune of transfer. If you clicked rune of transfer it will always say activate all lamps. Then when you activate a lamp thats when you find out the sub objective is